cocotb/share/lib/simulator/simulatormodule.cpp

```diff
@@ -661,16 +668,44 @@ static PyObject *set_signal_val_real(gpi_hdl_Object<gpi_sim_hdl> *self, PyObject
     Py_RETURN_NONE;
 }
 
-static PyObject *set_signal_val_long(gpi_hdl_Object<gpi_sim_hdl> *self, PyObject *args)
+static PyObject *set_signal_val_int(gpi_hdl_Object<gpi_sim_hdl> *self, PyObject *args)
 {
-    long value;
+    long long value;
     gpi_set_action_t action;
+    limits_e limits;
 
-    if (!PyArg_ParseTuple(args, "il:set_signal_val_long", &action, &value)) {
+    if (!PyArg_ParseTuple(args, "iLi:set_signal_val_int", &action, &value, &limits)) {
         return NULL;
     }
 
-    gpi_set_signal_value_long(self->hdl, value, action);
+    if (limits == limits_e::SIGNED_32BIT) {
```

Comment:
What's the motivation for handling this casting in C instead of Python?

Reply:
Your initial recommendation was to do the range checking in C++ for performance reasons. For this we need a type large enough to represent both int32_t and uint32_t, so using int64_t for the comparisons seemed the right solution.
After the range checking the value is cast down to 32 bit. Am I missing something here?